File: Android/tests/src/jp/co/riso/android/text/InvalidCharacterFilterTest.java

package jp.co.riso.android.text;

import android.text.InputFilter;
import android.widget.EditText;
import jp.co.riso.smartdeviceapp.SmartDeviceApp;
import junit.framework.TestCase;

public class InvalidCharacterFilterTest extends TestCase {
    private static final String INVALID_INPUT_TEXT = "!";
    private static final String VALID_INPUT_TEXT = "a";

    public InvalidCharacterFilterTest(String name) {
        super(name);
    }

    protected void setUp() throws Exception {
        super.setUp();
    }

    protected void tearDown() throws Exception {
        super.tearDown();
    }

    // ================================================================================
    // Tests - constructors
    // ================================================================================

    public void testConstructor_NullSet() {
        try {
            new InvalidCharacterFilter(null);
        } catch (NullPointerException e) {
            fail();
        }
    }

    public void testConstructor_ValidSet() {
        InvalidCharacterFilter filter = new InvalidCharacterFilter("abcd");
        assertNotNull(filter);
    }
    
    // ================================================================================
    // Tests - filter
    // ================================================================================

    public void testFilter() {
        EditText editText = new EditText(SmartDeviceApp.getAppContext());
        editText.setFilters(new InputFilter[] {
                new InputFilter.LengthFilter(8),
                new InvalidCharacterFilter("abcd")
        });
        editText.setText(INVALID_INPUT_TEXT);
        editText.setText(VALID_INPUT_TEXT);
    }
}

File: Android/app/src/androidTest/java/jp/co/riso/android/text/InvalidCharacterFilterTest.kt
package jp.co.riso.android.text

import android.text.InputFilter.LengthFilter
import android.widget.EditText
import jp.co.riso.smartdeviceapp.SmartDeviceApp
import junit.framework.TestCase.assertNotNull
import junit.framework.TestCase.fail
import org.junit.Test

class InvalidCharacterFilterTest {

    // ================================================================================
    // Tests - constructors
    // ================================================================================
    @Test
    fun testConstructor_NullSet() {
        try {
            InvalidCharacterFilter(null)
        } catch (e: NullPointerException) {
            fail()
        }
    }

    @Test
    fun testConstructor_ValidSet() {
        val filter = InvalidCharacterFilter("abcd")
        assertNotNull(filter)
    }

    // ================================================================================
    // Tests - filter
    // ================================================================================
    @Test
    fun testFilter() {
        val editText = EditText(SmartDeviceApp.getAppContext())
        editText.filters = arrayOf(
            LengthFilter(8),
            InvalidCharacterFilter("abcd")
        )
        editText.setText(INVALID_INPUT_TEXT)
        editText.setText(VALID_INPUT_TEXT)
    }

    companion object {
        private const val INVALID_INPUT_TEXT = "!"
        private const val VALID_INPUT_TEXT = "a"
    }
}